.github/workflows/build.yml

```diff
@@ -0,0 +1,100 @@
+---
+name: Build RPM packages
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_packages:
+    strategy:
+      matrix:
+        build_type:
+          - auto
+          - manual
+        version:
+          - 8
+          - 7
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+          architecture: "x64"
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build packages
+        run: make ${{ matrix.build_type }}${{ matrix.version }}
+
+      - name: Cache built packages
+        uses: actions/cache@v2
+        id: restore-build
+        with:
+          path: _dist${{ matrix.version }}/*
+          key: ${{ github.sha }}-${{ matrix.build_type }}${{ matrix.version }}
+
+  publish_packages:
+    needs: build_packages
+    if: github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        version:
+          - 8
+          - 7
```

Comment:
@lest, is there any point in separating publish jobs and run `make publish7` & `make publish8` separately instead of doing that in single step that includes both of them? (`make publish`)

Reply:
I think we can have a single step that includes both with the new approach for caching RPMs between jobs.